### PR TITLE
Fix for #440 - Removed GuiData class

### PR DESCRIPTION
**Pull request by toasterhead-master**
Fixes #440

### CODE REVIEW

**Comment by soraphis** [comment]
Not sure what the issue with the File Changes preview is, tbh.
I guess it's because #440 was squashed into a new merge commit and now this branch has trouble to find the nearest common ancestor.

which results in the conflict and the changes looking more complex than they have to be.

I guess the simplest solution would be to reset this branch (soft?) to canary. this should keep the file changes but adjust the commit history. then doing the "Removed GuiData" commit with the remaining changes.

**Comment by toasterhead-master** [comment]
> Not sure what the issue with the File Changes preview is

I'm just as confused as you are. I thought since my feature was pushed into the newest version then the `Files changed` would reflect that, but I guess I'm missing something here. If I had created a new fork, would we not have this problem?

**Comment by soraphis** [comment]
It's not about the fork, it's because the first PR was merged by squashing.

Lets say before #440 the last commit was X
So 440 introduced the commits A, B, ....., C on top of X and this PR introduces commit D on top of C

so git thinks to merge this PR it has to apply A, B, ...., C, D on top of X and make that the new header.
But in the meantime PR 440 was merged by squasing so the current git commit tree is X, Z (where Z contains all the changes from A, B, ...., C in a single commit).

Basically like this
```
                         │PR440
                         │       │PR448
                         ▼       ▼
       ┌────A──B────...──C───────D
       │
  ────X└────[     Z      ]
                  ▲
                  │Canary

```

So, you have to rebase D onto Z, or like i suggested create a new commit D' that contains the same changes as D but ontop of Z

**Comment by toasterhead-master** [comment]
Ok, I got it and fixed it, thanks for the help. I just needed to pull the changes from the recent 4.5.0 changes into my forked repo and then push my changes again for it to work, makes sense overall.

**Comment by soraphis** [comment]
Yeah, should work too. and will get squashed anyways when accepted. so should be fine

File: Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs [hunk 1]
```diff
@@ -79,53 +71,43 @@ public override float GetPropertyHeight(SerializedProperty property, GUIContent
 
             var innerProperty = property.FindPropertyRelative(usageData.PropertyName);
 
-            bool forceSingleLine = false;
```
Comment (soraphis) [inline Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs:82]: why was forceSingleLine deleted?
this is what @AdamRamberg  added as a fix in #447 I gues it got lost, due to the way this PR was created by overriding the changes from the previous PRs

Comment (soraphis) [inline Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs:82]: why was forceSingleLine deleted?
this is what @AdamRamberg  added as a fix in #447 I gues it got lost, due to the way this PR was created by overriding the changes from the previous PRs

**Review by soraphis — CHANGES_REQUESTED** [review]

File: Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs (hunk 1 above)
Comment (toasterhead-master) [reply Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs:82]: My bad, I should've checked if any external changes took place while my repo stayed the same. Fixing it

**Review by soraphis — APPROVED** [review]

File: Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs [hunk 2]
```diff
@@ -91,41 +85,38 @@ public override float GetPropertyHeight(SerializedProperty property, GUIContent
...
                     // In versions prior to 2022.3 GetPropertyHeight returns the wrong value for "SerializedPropertyType.Quaternion"
                     // In later versions, the fix is introduced _but only_ when using the SerializedPropertyType parameter, not when using the SerializedProperty parameter version.
                     // ALSO the SerializedPropertyType parameter version does not work with the isExpanded flag which we set to true exactly for this reason a (few) lines above.
-                    EditorGUI.GetPropertyHeight(SerializedPropertyType.Vector3, guiData.Label) :
-                    EditorGUI.GetPropertyHeight(usageTypeProperty, guiData.Label);
-#else
-                var valueFieldHeight = EditorGUI.GetPropertyHeight(usageTypeProperty, guiData.Label);
-#endif
+                    EditorGUI.GetPropertyHeight(SerializedPropertyType.Vector3, label) :
+                    EditorGUI.GetPropertyHeight(usageTypeProperty, label);
```
[the remark this answers is not in the document]
Comment (toasterhead-master) [reply Packages/Core/Editor/Drawers/AtomBaseReferenceDrawer.cs:119]: Oh thanks! Sorry, I didn't mean to remove it, don't know how I missed it or how it got removed :/

**Review by AdamRamberg — APPROVED** [review]
